### DISCUSSION TITLE
Simple way to set property mandatory only on insert

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#6861] [General]
Hi all,

I would like a simple way to define property required only when insert new record

is there a decoration for this? Something like

```
public sealed class xyzRow : Row<xyzRow.RowFields>, IIdRow, INameRow
{

    [DisplayName("Password"), **RequiredOnInsert(true)**]
    public string Password
    {
        get => fields.Password[this];
        set => fields.Password[this] = value;
    }
...
}
```

My problem is that I want password required only when insert data the first time.
When modify data, if the password field is empty, it means that user do not want to change it!

Thank you,
Elena.

**Comment by arunkumar12712** [comment]
[NotNull, Updatable(false)]

**Reply by elenaaralla** [reply]
Hi,

Updatable(false) set field non editable; I want the filed editable but non mandatory.

Thank you.

**Comment by VictorTomaili** [comment]
Check UserSaveHandler

https://github.com/serenity-is/serene/blob/master/src/Serene.Web/Modules/Administration/User/RequestHandlers/UserSaveHandler.cs#L84

**Reply by elenaaralla** [reply]
Thank you.

**Comment by elenaaralla** [comment]
Resolved with @VictorTomaili  example.
Thank you.
